research-software-directory/RSD-as-a-service
# Initial eslint vscode setup

# Eslint vscode setup

Based on [this article](https://www.aleksandrhovhannisyan.com/blog/format-code-on-save-vs-code-eslint/#3-formatting-on-save-in-vs-code-with-eslint)

How to test:
- If you do not have Eslint or TailwindCSS plugin installed VSCode should advice you to install it. I have not tested this feature (already have plugins installed).
- If you change some code in tsx or ts files, for example add ";" at the end of the import or spaces at the end of line of some code, VSCode should show warning (in orange).
- When you use save the file (Ctrl/Cmd+S), the code should be corrected, in our example ";" or spaces should be removed.
- When you want to assign TailwinCSS class, you should get suggestions what is possible to use. In addition, on save the classes will be ordered in the specific order (recommended by TailwindCSS).

In my case the first implementation did not work properly. After investigation I found that format on save approach will not work if AutoSave options is enabled in VSC. So in case of problems check if AutoSave is enabled.
